helm/app-operator-chart/templates/rbac.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ .Values.name }}
+  labels:
+    app: {{ .Values.name }}
+    giantswarm.io/service-type: "managed"
+rules:
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - "*"
+- apiGroups:
+    - application.giantswarm.io
+  resources:
+    - apps
+  verbs:
+    - get
+    - list
+    - patch
+    - update
+    - watch
+- apiGroups:
+    - core.giantswarm.io
+  resources:
+    - charts/status
+  verbs:
+    - create
+    - patch
+    - update
+- apiGroups:
+  - core.giantswarm.io
+  resources:
+  - charts
+  verbs:
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - core.giantswarm.io
+  resources:
+  - charts/status
+  verbs:
+  - create
+  - patch
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  - pods
+  verbs:
+  - "get"
+  - "list"
+- apiGroups:
```

Comment:
why do we need get and list on CMs and pods? afaik we do not have anything to do there in app operator, we might have a CM with the global config to configure it, but that should be a single CM not all, also we need access to appCatalog CRs

Reply:
We will need this later as app-operator will need to manage configmaps for chart-operator. But we should keep this as the minimal permissions for now and add it later when we need it.

Reply:
I will delete this permission at this moment

Reply:
ah, right, CMs (and maybe secrets at some point) it will need to create for charts at least for charts in Control Plane we would need that

Reply:
Yes for the chart CRs I think we should make sure it works for tenant and control plane clusters.

If the chart CR has a `kubeconfig` spec it will use it to access the tenant cluster. If its empty then it should create the chart CR in the control plane cluster the operator is running in.

Reply:
hmm, wait, right, it should have a kubeconfig also for control plane cluster, so it does not even need any of the chart related roles in here

Reply:
but shouldn't it also get a kubeconfig for control plane? I thin that would be nicer than splitting it between kubeconfig or SA RBAC

Reply:
hmm, ok, if you think that's nicer we can go that way, but then let's have notes/comments in thise RBAC yaml to point out which rules are for CP chart functionality, or better even let's split the RBAC role into one for normal app-operator functionality and one for applying chart CRs, that way we can have the latter both bound to the inCluster SA config on control plane but also bound in tenant clusters to the kubeconfig. WDYT? That way if we have changes, we just sync the two roles, for app-operator the role binding can just bind both roles.

Reply:
In the rbac.yaml we should have 2 cluster roles, `app-operator` and `app-operator-chart`. The `-chart` role should just have the permissions for managing chart CRs.

The reason for this is when managing chart CRs inside tenant clusters we will use the `kubeconfig` rather than the `-chart` role.